Src/Plugins/Security/Security.SharedStrings.txt

```diff
@@ -45,6 +45,9 @@
   $SEC101/016.StripeApiKey=\b(?<refine>(?<key>(?:r|s)k_(live|test)_(?i)[0-9a-z]{24,99}))([^0-9a-z]|$)
   $SEC101/017.NpmAuthorToken=(?i)npm.{0,100}[^0-9a-z](?-i)(?<refine>(?<key>[a-z0-9]{8}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{12}))[^0-9A-Za-z]
   $SEC101/019.PicaticApiKey=\b(?<refine>sk_(live|test)_[0-9a-z]{32})(?i)([^0-9a-z]|$)
+  $SEC101/020.DropboxAccessTokenNoExpiration=(?si)dropbox.{1,50}[^0-9a-z](?<refine>[0-9a-z_-]{64})([^0-9a-z]|$)
+  $SEC101/020.DropboxAccessTokenShortExpiration=(?si)dropbox.{1,50}\b(?<refine>sl\.(?i)[0-9a-z-_]{135,139})(?i)([^0-9a-z]|$)
```

Comment:
>DropboxAccessTokenShortExpiration [](start = 14, length = 33)

It'd be nice if we could distingish a secret that never expires from other secrets. Is that possible for us? IOW, it is a more serious concern to flag a key that never expires than one that is fixed in time-frame (in terms of validity).

Reply:
The only way to do that is checkin the size. no expiration has a fixed size (64) characters. different from that  is short lived.
i could also  make a check and see if its a short lived, return as authorized warning
otherwise, authorized error

---
In reply to: [598009225](https://github.com/microsoft/sarif-pattern-matcher/pull/315#discussion_r598009225) [](ancestors = 598009225)